app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterView.kt

```diff
@@ -7,47 +7,45 @@ import com.puzzlebench.clean_marvel_kotlin.domain.model.Character
 import com.puzzlebench.clean_marvel_kotlin.presentation.MainActivity
 import com.puzzlebench.clean_marvel_kotlin.presentation.adapter.CharacterAdapter
 import com.puzzlebench.clean_marvel_kotlin.presentation.extension.showToast
+import com.puzzlebench.clean_marvel_kotlin.presentation.fragment.DetailCharacterFragmentDialog
 import kotlinx.android.synthetic.main.activity_main.*
 import java.lang.ref.WeakReference
 
-class CharecterView(activity: MainActivity) {
+class CharacterView(activity: MainActivity) {
     private val activityRef = WeakReference(activity)
     private val SPAN_COUNT = 1
-    var adapter = CharacterAdapter { character -> activity.applicationContext.showToast(character.name) }
-
+    private var adapter: CharacterAdapter? = null
     fun init() {
         val activity = activityRef.get()
         if (activity != null) {
+                adapter = CharacterAdapter { character ->
+                val fragment = DetailCharacterFragmentDialog.newInstance(character.id)
+                fragment.show(activity.fragmentManager,"Detail")
+            }
             activity.recycleView.layoutManager = GridLayoutManager(activity, SPAN_COUNT)
             activity.recycleView.adapter = adapter
             showLoading()
         }
-
     }
 
     fun showToastNoItemToShow() {
-        val activity = activityRef.get()
-        if (activity != null) {
-            val message = activity.baseContext.resources.getString(R.string.message_no_items_to_show)
-            activity.applicationContext.showToast(message)
-
-        }
+        val message = activityRef.get()?.baseContext?.resources?.getString(R.string.message_no_items_to_show)
```

Comment:
You don't need to get the baseContext to get the resources. You can get it from activity.